engine/src/main/java/org/pentaho/di/trans/steps/mysqlbulkloader/MySQLBulkLoader.java

```diff
@@ -0,0 +1,629 @@
+                data.fifoStream.write( valueMeta.getBinaryString( valueData ) );
+              } else {
+                Date date = valueMeta.getDate( valueData );
+                if ( date != null ) {
+                  data.fifoStream.write( data.bulkFormatMeta[i].getString( date ).getBytes() );
+                }
+              }
+              break;
+            case ValueMetaInterface.TYPE_BOOLEAN:
+              if ( valueMeta.isStorageBinaryString() && data.bulkFormatMeta[i] == null ) {
+                data.fifoStream.write( valueMeta.getBinaryString( valueData ) );
+              } else {
+                Boolean b = valueMeta.getBoolean( valueData );
+                if ( b != null ) {
+                  data.fifoStream.write( data.bulkFormatMeta[i].getString( b ).getBytes() );
+                }
+              }
+              break;
+            case ValueMetaInterface.TYPE_NUMBER:
+              if ( valueMeta.isStorageBinaryString() && data.bulkFormatMeta[i] == null ) {
+                data.fifoStream.write( (byte[]) valueData );
+              } else {
+                /**
+                 * If this is the first line, reset default conversion mask for Number type (#.#;-#.#).
+                 * This will make conversion mask to be calculated according to meta data (length, precision).
+                 *
+                 * http://jira.pentaho.com/browse/PDI-11421
+                 */
+                if ( getLinesWritten() == 0 ) {
+                  data.bulkFormatMeta[i].setConversionMask( null );
+                }
+
+                Double d = valueMeta.getNumber( valueData );
+                if ( d != null ) {
+                  data.fifoStream.write( data.bulkFormatMeta[i].getString( d ).getBytes() );
+                }
+              }
+              break;
+            case ValueMetaInterface.TYPE_BIGNUMBER:
+              if ( valueMeta.isStorageBinaryString() && data.bulkFormatMeta[i] == null ) {
+                data.fifoStream.write( (byte[]) valueData );
+              } else {
+                BigDecimal bn = valueMeta.getBigNumber( valueData );
+                if ( bn != null ) {
+                  data.fifoStream.write( data.bulkFormatMeta[i].getString( bn ).getBytes() );
+                }
+              }
+              break;
+            default:
+              break;
+          }
+        }
+      }
+
+      // finally write a newline
+      //
+      data.fifoStream.write( data.newline );
+
+      if ( ( getLinesOutput() % 5000 ) == 0 ) {
+        data.fifoStream.flush();
+      }
+    } catch ( IOException e ) {
+      // If something went wrong with writing to the fifo, get the underlying error from MySQL
+      try {
+        logError( BaseMessages.getString( PKG, "MySQLBulkLoader.Message.IOERROR", this.threadWaitTimeText ) );
+        try {
+          data.sqlRunner.join( this.threadWaitTime );
+        } catch ( InterruptedException ex ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)